src/services/factories/tokenList.ts

```diff
@@ -88,32 +99,124 @@ export function getTokensFactory(factoryParams: {
     // if something failed...
     if (failedToUpdate.length > 0) {
       logDebug(
-        `[network:${networkId}] Failed to fetch ids for ${failedToUpdate.length} tokens. Trying again in ${retryDelay}ms`,
+        `[tokenListFactory][${networkId}] Failed to fetch ids for ${failedToUpdate.length} tokens. Trying again in ${retryDelay}ms`,
       )
       // calculate how long to wait next time
       const nextDelay = retryDelay * (exponentialBackOff ? 2 : 1)
       // try again failed tokens after retryDelay
-      setTimeout(() => updateWithRetries(networkId, failedToUpdate, maxRetries - 1, nextDelay), retryDelay)
+      setTimeout(() => updateTokenIds(networkId, failedToUpdate, maxRetries - 1, nextDelay), retryDelay)
     }
   }
 
-  async function updateTokenIds(networkId: number): Promise<void> {
-    // Set as updated on start to prevent concurrent queries
-    updatedIdsForNetwork.add(networkId)
+  async function getErc20DetailsOrAddress(networkId: number, tokenAddress: string): Promise<TokenFromErc20 | string> {
+    // Simple wrapper function to return original address instead of null for make logging easier
+    const erc20Details = await getTokenFromErc20({ networkId, tokenAddress })
```

Comment:
Careful what `erc20Details` returns, is it possible that this promise always resolves to something `truthy`? then this fn will fail on `114` in the sense that `tokenAddress` will never return - but then I guess TS would catch this?

Reply:
`getTokenFromErc20` returns `null` when the token is not a valid Erc20.

This function just wraps the response to, instead of null, return the token address so we can log which token failed to load.